locals.tf

```diff
@@ -1,16 +1,16 @@
 locals {
-  # Get a name from the descriptor. If not available, use default naming convention.
-  # Trim and replace function are used to avoid bare delimiters on both ends of the name and situation of adjacent delimiters.
-  name_from_descriptor = module.database_label.enabled ? trim(replace(
-    lookup(module.database_label.descriptors, var.descriptor_name, module.database_label.id), "/${module.database_label.delimiter}${module.database_label.delimiter}+/", module.database_label.delimiter
-  ), module.database_label.delimiter) : null
+  context_template = lookup(var.context_templates, var.name_scheme.context_template_name, null)
 
-  create_default_roles = module.this.enabled && var.create_default_roles
+  default_role_naming_scheme = {
+    properties            = ["name"]
+    context_template_name = "snowflake-database-database-role"
```

Comment:
you need unique name to distinct database role for a database so double `database` is expected right?

Reply:
Yup and additionally I tend to follow the `snowflake-MODULE-ROLE_TYPE` schema, so there is also `snowflake-schema-database-role` and `snowflake-stage-database-role` so IMO it is only logical to go with `snowflake-database-database-role` :smiley: